language/tools/move-mv-llvm-compiler/src/stackless/module_context.rs

```diff
@@ -848,6 +848,84 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
         attrs
     }
 
+    // This function extracts an entry function actual arguments from
+    // AccountInfo data field.  The field is a byte array containing
+    // values of actual arguments in sequential order without gaps.
+    fn emit_entry_arguments(
+        &self,
+        fn_env: &mm::FunctionEnv,
+        accounts: &llvm::AnyValue,
+    ) -> Vec<llvm::AnyValue> {
+        if fn_env.get_parameter_count() == 0 {
+            return vec![];
+        }
+        let llcx = self.llvm_cx;
+        let vec_ty = llcx.get_anonymous_struct_type(&[
+            llcx.ptr_type(),
+            llcx.int_type(64),
+            llcx.int_type(64),
+        ]);
+        let acc_vec_ptr =
+            self.llvm_builder
+                .getelementptr(*accounts, &vec_ty.as_struct_type(), 0, "acc_vec_ptr");
+        let acc_vec_ptr =
+            self.llvm_builder
+                .load(acc_vec_ptr, llcx.ptr_type(), "acc_vec_ptr_loaded");
+        let data_ty = llcx.get_anonymous_struct_type(&[llcx.ptr_type(), llcx.int_type(64)]);
+        let acc_ty = llcx.get_anonymous_struct_type(&[
+            llcx.ptr_type(),   // key
+            llcx.int_type(64), // lamports
+            data_ty,           // data
+            llcx.ptr_type(),   // owner
+            llcx.int_type(64), // rent_epoch
+            llcx.int_type(1),  // is_signer
+            llcx.int_type(1),  // is_writable
+            llcx.int_type(1),  // executable
+        ]);
+        let acc_data =
+            self.llvm_builder
+                .getelementptr(acc_vec_ptr, &acc_ty.as_struct_type(), 2, "acc_data");
+        let acc_data_ptr =
+            self.llvm_builder
+                .getelementptr(acc_data, &data_ty.as_struct_type(), 0, "acc_data_ptr");
+        let acc_data_ptr =
+            self.llvm_builder
+                .load(acc_data_ptr, llcx.ptr_type(), "acc_data_ptr_loaded");
+        // Construct a struct of fields with types corresponding to
+        // the entry function formal parameters.  For reference
+        // parameters, use the type it refers to.  The struct helps to
+        // compute offsets in acc_data slice for accessing actual
+        // argument values.
```

Comment:
Very nice idea-- clean and elegant.

Reply:
Should mutable reference parameters to an entry function be considered as capable of modifying the global store (i.e. the ledger database)?  I doubt anything other than accounts (ie address or signer parameters) can be written to in Solana ledger... Accounts are modifiable only is is_writable field is set to true in AccountInfo for the account.

Reply:
That is one of the same questions I had. I really meant we, as a group, need to understand what are the semantics. And it may not match exactly what is legal on the other Move-based networks due to Solana's global store model.